mne_connectivity/base.py

```diff
@@ -0,0 +1,548 @@
+                raise RuntimeError(f'Data not using an epoched data '
+                                   f'structure should have at least '
+                                   f'1 dimensions and at most 3 '
+                                   f'dimensions. Your data was '
+                                   f'{data.shape} shape.')
+
+        # check that the indices passed in are of the same length
+        if isinstance(self.indices, tuple):
+            if len(self.indices[0]) != len(self.indices[1]):
+                raise ValueError(f'If indices are passed in '
+                                 f'then they must be the same '
+                                 f'length. They are right now '
+                                 f'{len(self.indices[0])} and '
+                                 f'{len(self.indices[1])}.')
+
+    @property
+    def _data(self):
+        """Numpy array of connectivity data."""
+        return self.xarray.values
+
+    @property
+    def dims(self):
+        """The dimensions of the xarray data."""
+        return self.xarray.dims
+
+    @property
+    def coords(self):
+        """The coordinates of the xarray data."""
+        return self.xarray.coords
+
+    @property
+    def attrs(self):
+        """Attributes of connectivity dataset.
+
+        See ``xarray``'s ``attrs``.
+        """
+        return self.xarray.attrs
+
+    @property
+    def shape(self):
+        """Shape of raveled connectivity."""
+        return self.xarray.shape
+
+    @property
+    def names(self):
+        """Node names."""
+        return self.attrs['node_names']
+
+    @property
+    def xarray(self):
+        """Xarray of the connectivity data."""
+        return self._obj
+
+    @property
+    def n_epochs_used(self):
+        """Number of epochs used in computation of connectivity.
+
+        Can be 'None', if there was no epochs used.
+        """
+        return self.attrs.get('n_epochs_used')
+
+    @property
+    def _size(self):
+        """Estimate the object size."""
+        size = 0
+        size += object_size(self.get_data())
+        size += object_size(self.attrs)
+        return size
+
+    def get_data(self, output='raveled', squeeze=True):
+        """Get connectivity data as a numpy array.
+
+        Parameters
+        ----------
+        output : str, optional
+            How to format the output, by default 'raveled', which
+            will represent each connectivity matrix as a
+            ``(n_nodes_in * n_nodes_out, 1)`` list. If 'full', then
+            will return each connectivity matrix as a 2D array.
+        squeeze : bool, optional
+            Whether to squeeze the array or not, by default True.
+
+        Returns
+        -------
+        data : np.ndarray
+            The output connectivity data.
+        """
+        if output not in ['raveled', 'full']:
+            raise ValueError(f'Output of data can only be one of '
+                             f'"raveled, "full", not {output}.')
```

Comment:
use the `from mne.utils import _check_option` and
```suggestion
        _check_option('output', output, ['raveled', 'full'])
```